tests/test_config/test_config_files.py

```diff
@@ -35,3 +36,20 @@ def test_load_config_file(ext, handler):
     d = load_config_file(f_name)
     assert d == data
     os.remove(f_name)
+
+
+def test_load_config_file_exception_msg_invalid_ext():
+    handle, f_name = tempfile.mkstemp(suffix='.invalid')
+    try:
+        f = os.fdopen(handle, "w")
```

Comment:
Why exactly is opening and closing the file here necessary? As it is not really obvious, a comment would help here. Otherwise we might also go with 
```
    _, f_name = tempfile.mkstemp(suffix='.invalid')
    try:
        exception_msg = re.compile(
            'Configuration file ".*.invalid" has invalid or '
            'unsupported extension ".invalid".'
        )
        with pytest.raises(ValueError) as excinfo:
            load_config_file(f_name)
        assert exception_msg.match(excinfo.value.args[0])
    finally:
        os.remove(f_name)
``` 
which seems to work as well.

Reply:
You are right. I copied the example from the test above to be compatible with Python2.
I fixed it.